lib/3scale_toolbox/commands/remote_command.rb

```diff
@@ -0,0 +1,61 @@
+require 'cri'
+require '3scale_toolbox/base_command'
+require '3scale_toolbox/commands/remote_command/remote_add'
+require '3scale_toolbox/commands/remote_command/remote_remove'
+require '3scale_toolbox/commands/remote_command/remote_rename'
+
+module ThreeScaleToolbox
+  module Commands
+    module RemoteCommand
+      class RemoteCommand < Cri::CommandRunner
+        include ThreeScaleToolbox::Command
+        def self.command
+          Cri::Command.define do
+            name        'remote'
+            usage       'remote <command> [options]'
+            summary     '3scale CLI remote'
+            description '3scale CLI command to manage your remotes'
+            runner RemoteCommand
+          end
+        end
+
+        def run
+          list_remotes
+        rescue StandardError => e
+          warn e.message
+          exit 1
+        end
+
+        def invalid_remote
+          raise "invalid remote configuration from config file #{config_file}"
+        end
+
+        def validate_remotes(remotes)
+          return if remotes.nil?
+          invalid_remote unless remotes.class == Hash
+          remotes.each_value do |remote|
+            invalid_remote unless remote.key?(:endpoint) && remote.key?(:provider_key)
+          end
+        end
+
+        def list_remotes
+          remotes = config.data :remotes
+
+          validate_remotes(remotes)
```

Comment:
I think the exception should not be thrown from inside the validation, but rather from here.


```ruby
return invalid_remote unless valid_remotes?
```

And `valid_remotes?` would just return `true/false`. There is no need to raise exceptions unless in exceptional situation.